lib/main.js

```diff
@@ -87,6 +91,11 @@ export default {
           args.push("-xc++");
           args.push(...atom.config.get("linter-clang.clangDefaultCppFlags").split(/\s+/));
         }
+        if(activeEditor.getGrammar().name === "C++14") {
+          //const language = "c++";
+          args.push("-xc++");
+          args.push(...atom.config.get("linter-clang.clangDefaultCpp14Flags").split(/\s+/));
```

Comment:
It works in the settings page, trust me, I work with atom packages each day :)
You do the change I suggested above and then you'll get an array when you get this config.